src/librustc_trans/trans/cleanup.rs

```diff
@@ -769,20 +771,24 @@ impl<'blk, 'tcx> CleanupHelperMethods<'blk, 'tcx> for FunctionContext<'blk, 'tcx
                 }
             }
 
+            // Pop off the scope, since we may be generating
+            // unwinding code for it.
+            popped_scopes.push(self.pop_scope());
```

Comment:
The first suggestion would drop the scope with the cached exit from the set of cleanup scopes. There's a `break` in the `if let`, so it would never be pushed (unless I duplicated the push). I'll go with the second one.